### DISCUSSION TITLE
Animated .webp support?

### DISCUSSION THREAD

**Discussion by GuitarBro** [ludumdare/ludumdare#2162] [Ideas]
Given the better compression than animated .gif images, it seems like animated .webp support would be a welcome change from both an uploader and viewer standpoint.

As far as I can tell from a cursory glance at the upload code, there's no reason that animated webp would be inherently impossible (unless I'm missing something). Unfortunately, it appears that php's getimagesize() fails on a webp image if running a version older than 7.1 (which, I assume is the case for the ldjam site?) which causes the upload to be rejected.

I'm not sure what it would take to get this to work (if it involves a php upgrade, probably more work than this feature alone is worth) but I figured I'd at least leave this here to indicate that there is some interest in using it.

**Comment by mikekasprzak** [comment]
I'm actually looking to remove/replace our local image processing since it's a potential attack vector. But yes, GIF is an outdated animation format. Transcoding GIF to MP4 is one of the more popular options, since mobile devices often have hardware h.264 decoders. That not only makes them smaller but it saves battery.

Animated webp is not something I've looked closely at, its browser compatibility, but my main concern would be mobile battery usage. That and webp isn't something that's well supported by traditional image and video editing software. It's common to see jpeg and png files converted to webp by servers, but webp is rarely an authoring format used by a client.

**Reply by GuitarBro** [reply]
That's fair, webp is certainly not the most common from an authoring standpoint (though I believe browser support is now quite good). 

Personally, .mp4 would actually be even more preferable since I happen to record videos in that format already, I just assumed that another image format would be less hassle to implement than supporting video as well.
